### PR TITLE
fix(textarea): stacked/floating textarea size is correct on safari

### LINKED ISSUE #27345: bug: ion-textarea is broken on IOS if label position "floating" or "stacked" is used
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

If the ion-textarea is used on safari with the label position floating or stacked, the text goes outside the textarea and prevents elements behind from being clickable.

image

### Expected Behavior

I expect the same behavior as when I use the textarea with another label position (start). 

image

### Steps to Reproduce

1. create new Textare with a label and label position floating or stacked
2. open the textarea in Safari and start adding multiple rows

### Code Reproduction URL

https://github.com/bhalbritter/IonicTextareaBug

### Ionic Info

Ionic:

   Ionic CLI       : 7.0.1 (/opt/homebrew/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/react 7.0.4

Capacitor:

   Capacitor CLI      : 4.8.0
   @capacitor/android : not installed
   @capacitor/core    : 4.8.0
   @capacitor/ios     : not installed

Utility:

   cordova-res : 0.15.4
   native-run  : 1.7.2

System:

   NodeJS : v19.2.0 (/opt/homebrew/Cellar/node/19.2.0/bin/node)
   npm    : 9.6.0
   OS     : macOS Unknown

### Additional Information

Safari Version 16.3 (18614.4.6.1.6)

On Chrome everything works fine.

### PR DESCRIPTION
**Pull request by liamdebeasi**
Issue number: resolves #27345

---------

## What is the current behavior?

WebKit has a bug where children of grid elements placed inside flex elements have the wrong size. This is causing textarea's dimensions to spill out of its grid container. Grid is used for autogrow functionality, and the grid itself is placed inside `.label-wrapper` which uses flexbox for aligning the label and control.

## What is the new behavior?

- Adds `grid-auto-rows: 100%` to avoid the WebKit bug which specifies the size of implicitly-created grid rows.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Dev build: `7.1.3-dev.11688748118.18ff9e7d`
WebKit bug: https://bugs.webkit.org/show_bug.cgi?id=256781

### CODE REVIEW

**Comment by liamdebeasi** [comment]
Going to get a couple reviewers since this has to do with a WebKit bug.

File: core/src/components/textarea/textarea.scss [hunk 1]
```diff
@@ -323,6 +323,8 @@
   min-height: inherit;
   max-height: inherit;
 
+  grid-auto-rows: 100%;
```
Comment (amandaejohnston) [inline core/src/components/textarea/textarea.scss:326]: Can we add a comment noting that this is to fix a Webkit bug, and/or include a tech debt ticket to remove it once the bug is fixed?

**Review by amandaejohnston — APPROVED** [review]

File: core/src/components/textarea/textarea.scss (hunk 1 above)
Comment (liamdebeasi) [reply core/src/components/textarea/textarea.scss:333]: Done! https://github.com/ionic-team/ionic-framework/pull/27766/commits/d8c1b171084effaf2a6d662b74e94e386f00cbe8

Comment (thetaPC) [reply core/src/components/textarea/textarea.scss:333]: I agree with this comment.

**Review by thetaPC — APPROVED** [review]
LGTM

**Pull request merged**